packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js

```diff
@@ -1266,6 +1266,39 @@ module('Date AM/PM Handling', setupModule, () => {
     });
 });
 
+module('TimeZone Handling', setupModule, () => {
+    test('should support \'x\' in date pattern and not generate errors (T1241387)', function(assert) {
+        try {
+            this.instance.option({
+                displayFormat: 'yyyy-MM-dd\'T\'HH:mm:ssxxx',
+                useMaskBehavior: true,
+                type: 'date',
+            });
+            assert.ok(true, 'no error shown');
+        } catch(e) {
+            assert.ok(false, 'error exists');
+        }
+    });
+    test('should not show error when changing timezone on runtime via up/down buttons (T1241387)', function(assert) {
+        try {
+            this.instance.option({
+                displayFormat: 'yyyy-MM-dd\'T\'HH:mm:ssxxx',
+                useMaskBehavior: true,
+                type: 'date',
+            });
+            const oldValue = this.instance.option('value');
+            this.keyboard.caret(24);
+            this.$input.focus().trigger('dxclick');
+            this.keyboard.press('up');
+
+            assert.ok(true, 'no error shown');
+            assert.strictEqual(this.instance.option('value'), oldValue, 'value has not been modified');
```

Comment:
I think that we need to check input.val here instead of value because no "change" event was triggered => value cannot be changed

Reply:
[7360ba9](https://github.com/DevExpress/DevExtreme/pull/27770/commits/7360ba926fdbd2e9053e5d0339f7f948e847bfcf)